code/game/jobs/faction/faction.dm

```diff
@@ -57,10 +57,13 @@
 	if (!is_type_in_typecache(S, allowed_species_types))
 		return "Invalid species selected."
 
+	if(!is_visible(user))
+		return "This faction is not available to you."
```

Comment:
```suggestion
		return "You are not authorized to select this faction."
```

Reply:
The error message is intentionally generic as other factions might not be visible to the user in the future (outside of the admin faction)